src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,30 +1,78 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
-
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
-
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button data-cy="clearSearchButton" type="button" className="delete" />
-      </span>
-    </p>
-  </form>
-);
+import React, { useState } from 'react';
+
+type Props = {
+  onSelectedFilter: (filter: SortType) => void;
+  onQuery: (query: string) => void;
+};
+
+export enum SortType {
+  All = 'all',
+  Active = 'active',
+  Completed = 'completed',
+}
+
+export const TodoFilter: React.FC<Props> = ({
+  onSelectedFilter = () => {},
+  onQuery = () => {},
+}) => {
+  const [query, setQuery] = useState('');
```

Comment:
you don't need this state here, just pass the query value from the App to TodoFilter as a prop